packages/platform/types/bookings/2024-08-13/inputs/update-booking.input.ts

```diff
@@ -0,0 +1,39 @@
+import { ApiPropertyOptional, getSchemaPath } from "@nestjs/swagger";
+import { Type } from "class-transformer";
+import { IsOptional } from "class-validator";
+
+import {
+  BookingInputAddressLocation_2024_08_13,
+  BookingInputAttendeeAddressLocation_2024_08_13,
+  BookingInputAttendeeDefinedLocation_2024_08_13,
+  BookingInputAttendeePhoneLocation_2024_08_13,
+  BookingInputIntegrationLocation_2024_08_13,
+  BookingInputLinkLocation_2024_08_13,
+  BookingInputPhoneLocation_2024_08_13,
+  BookingInputOrganizersDefaultAppLocation_2024_08_13,
+  ValidateBookingLocation_2024_08_13,
+} from "./location.input";
+import type { BookingInputLocation_2024_08_13 } from "./location.input";
+
+export class UpdateBookingInput_2024_08_13 {
+  @IsOptional()
+  @ValidateBookingLocation_2024_08_13()
+  @ApiPropertyOptional({
+    description:
+      "One of the event type locations. If instead of passing one of the location objects as required by schema you are still passing a string please use an object.",
+    oneOf: [
+      { $ref: getSchemaPath(BookingInputAddressLocation_2024_08_13) },
+      { $ref: getSchemaPath(BookingInputAttendeeAddressLocation_2024_08_13) },
+      { $ref: getSchemaPath(BookingInputAttendeeDefinedLocation_2024_08_13) },
+      { $ref: getSchemaPath(BookingInputAttendeePhoneLocation_2024_08_13) },
+      { $ref: getSchemaPath(BookingInputIntegrationLocation_2024_08_13) },
+      { $ref: getSchemaPath(BookingInputLinkLocation_2024_08_13) },
+      { $ref: getSchemaPath(BookingInputPhoneLocation_2024_08_13) },
+      { $ref: getSchemaPath(BookingInputOrganizersDefaultAppLocation_2024_08_13) },
+    ],
+  })
+  @Type(() => Object)
+  // note(Rajiv): string is for backwards compatability
```

Comment:
The reason we still allowed string when creating a booking because we first only allowed passing a string, and only then introduced `BookingInputLocation_2024_08_13` and in create booking location we have `BookingInputLocation_2024_08_13 | string;` for backwards comaptability - meaning that if we did not have `| string` it would break existing requests containing location.

But since you are adding the update endpoint for the first time it should only accept `BookingInputLocation_2024_08_13` and there is no need to then check if typeof location is object or not - it has to be.